### PR TITLE
Add Data Migration documentation to docusite

### LINKED ISSUE #2975: Add documentation for data migration configuration
**Describe the Issue**

The migration functionality has been implemented in the [DataMigration](https://github.com/opensrp/fhircore/blob/main/android/quest/src/main/java/org/smartregister/fhircore/quest/data/DataMigration.kt) class.

The main configurations classes are [DataMigrationConfiguration.kt](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/migration/DataMigrationConfiguration.kt)  and [MigrationConfig.kt](https://github.com/opensrp/fhircore/blob/main/android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/migration/MigrationConfig.kt)

Sample config

```
{
  "appId": "echis",
  "configType": "dataMigration",
  "migrations": [
    {
      "resourceType": "Task",
      "updateValues": [
        {
          "jsonPathExpression": "Task.basedOn[0].reference",
          "valueRule": {
            "name": "taskBasedOn",
            "condition": "true",
            "actions": [
              "data.put('taskBasedOn', 'CarePlan/' + fhirPath.extractValue(Task, 'Task.basedOn[0].reference') )"
            ]
          }
        }
      ],
      "dataQueries": [
        {
          "paramName": "status",
          "operation": "OR",
          "filterCriteria": [
            {
              "dataType": "CODE",
              "value": {
                "system": "http://hl7.org/fhir/task-status",
                "code": "ready"
              }
            },
            {
              "dataType": "CODE",
              "value": {
                "system": "http://hl7.org/fhir/task-status",
                "code": "requested"
              }
            },
            {
              "dataType": "CODE",
              "value": {
                "system": "http://hl7.org/fhir/task-status",
                "code": "in-progress"
              }
            }
          ]
        }
      ],
      "purgeAffectedResources": true,
      "resourceFilterExpression" : {
        "conditionalFhirPathExpressions": [
          "Task.basedOn[0].reference.startsWith('CarePlan').not() and Task.basedOn[0].reference.startsWith('PlanDefinition').not()"
        ],
        "matchAll": true
      },
      "version": 1
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by sharon2719**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #2975 

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [x] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

**Comment by pld** [comment]
I'm moving this to the configurations section

File: docs/engineering/android-app/configuring/data-migration.mdx [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Data migration
+
+
+The main functionalities of data migration are:
+- Migrating data by updating resources based on a set of data queries and key-value pairs,
+- Retrieving the latest migration version from shared preferences,
+- Performing the migration only if the new version is greater than or equal to the client's latest stored migration version.
+
+The data migration configuration outlines a set of rules and operations for migrating FHIR resources, specifically `Task` resource migration. With this configuration one is able to filter the target `Task`, read its referenced `PlanDefinition` and `rev_including` its careplans then filter for the task that lies within the careplan period and read the resulting careplan id then use it to update the `task.basedOn`. It ensures that all the resources that are being generated through `StructureMap`s are not malformed.
```
Comment (pld) [inline docs/engineering/android-app/configuring/data-migration.mdx:9]: Some questions on this:

1. Do migrations only work for `Task`s or is this an example and they can be applied to other resources as well?
2. What does `rev_including` mean?
3. Why does it need to update the `task.basedOn` field?
4. How does it ensure that the resource generated by SMs are not malformed?

**Review by pld — DISMISSED** [review]

File: docs/engineering/android-app/configuring/data-migration.mdx (hunk 1 above)
Comment (ageryck) [reply docs/engineering/android-app/configuring/data-migration.mdx:9]: 1. Task should be just an example resource targetted by the migration scripts written
2. In FHIR Search API Forward inclusion with _include adds resources and resource versions referenced by the primary results, and reverse inclusion with _revinclude adds resources that reference the primary results.
3. Again this is just an example resource `attribute` targetted by the migration scripts written it can be any attribute within the resource, only noting that currently, we can only update an already existing attribute and not create a new attribute
4. This is not true, @sharon2719 lets remove the statement or reword it to` it cleans out the malformed resource attributes that a wrong StructureMap may have generated`

File: docs/engineering/android-app/configuring/data-migration.mdx [hunk 2]
```diff
@@ -1,12 +1,12 @@
-# Data Migration
+# Data migration
```
Comment (ageryck) [inline docs/engineering/android-app/configuring/data-migration.mdx:1]: reword this sections based on suggested notes

File: docs/engineering/android-app/configuring/data-migration.mdx [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+- Performing the migration only if the new version is greater than or equal to the client's latest stored migration version.
 
-The data migration configuration outlines a set of rules and operations for migrating FHIR resources , specifically `Tasks`migration. With this configuration one is able to filter the target task, read its referenced `plandef` and `rev_including` its careplans then filter for the task that lies within the careplan period and read the resulting careplan id then use it to update the `task.basedOn`.It ensures that all the resources that are being generated through structuremaps are not malformed.
+The data migration configuration outlines a set of rules and operations for migrating FHIR resources, specifically `Task` resource migration. With this configuration one is able to filter the target `Task`, read its referenced `PlanDefinition` and `rev_including` its careplans then filter for the task that lies within the careplan period and read the resulting careplan id then use it to update the `task.basedOn`. It ensures that all the resources that are being generated through `StructureMap`s are not malformed.
 
 ## Sample JSON
```
Comment (ageryck) [inline docs/engineering/android-app/configuring/data-migration.mdx:11]: You can include a sample with a few rules, and mention the `rules` and `data queries` work exactly the same as they are used on the larger product

**Review by ageryck — CHANGES_REQUESTED** [review]

**Review by ageryck — APPROVED** [review]
